Sources/Public/DotLottie/Utility/DotLottieImageProvider.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  DotLottieImageProvider.swift
+//  Lottie
+//
+//  Created by Evandro Hoffmann on 20/10/22.
+//
+
+import Foundation
+#if os(iOS) || os(tvOS) || os(watchOS) || targetEnvironment(macCatalyst)
+import UIKit
+#elseif os(macOS)
+import AppKit
+#endif
+
+// MARK: - DotLottieImageProvider
+
+/// Provides an image for a lottie animation from a provided Bundle.
+public class DotLottieImageProvider: AnimationImageProvider {
+
+  // MARK: Lifecycle
+
+  /// Initializes an image provider with a specific filepath.
+  ///
+  /// - Parameter filepath: The absolute filepath containing the images.
+  ///
+  public init(filepath: String) {
+    self.filepath = URL(fileURLWithPath: filepath)
+    loadImages()
+  }
+
+  public init(filepath: URL) {
+    self.filepath = filepath
+    loadImages()
+  }
+
+  // MARK: Public
+
+  public func imageForAsset(asset: ImageAsset) -> CGImage? {
+    imageCache.object(forKey: asset.name as NSString)
+  }
+
+  // MARK: Internal
+
+  let filepath: URL
+
+  // MARK: Private
+
+  private var imageCache: NSCache<NSString, CGImage> = .init()
```

Comment:
In this case I think we actually _shouldn't_ use `NSCache`.

The system will remove entires from an `NSCache` when under memory pressure. In our case we can't reload these images later, since the files no longer exists.

```suggestion
  private var images: [String: CGImage] = [:]
```